internal/isoeditor/rhcos.go

```diff
@@ -430,3 +436,71 @@ func writeOffsetInfo(headerOffset int64, offsetInfo *OffsetInfo, iso *os.File) (
 
 	return bytesLength, nil
 }
+
+// GetEmbedArea takes the 24 bytes of a CoreOS ISO and validates it is a
+// CoreOS ISO file before returning the offset into the file (and length) of
+// the embed area.
+func GetEmbedArea(header []byte) (int64, int64, error) {
+	// NOTE: The last 24 bytes should be:
+	// 8 bytes: magic string "coreiso+" tells us this is a CoreOS ISO image
+	// 8 bytes little-endian: offset of embed area from start of ISO file
+	// 8 bytes little-endian: length of embed area
+	// ref: https://github.com/coreos/coreos-installer/blob/80b828caa037bf77968fb472dffe63fe77e15826/src/live.rs#L435
+	if len(header) != 24 {
+		return 0, 0, errors.Errorf("Header unexpected size %v, expected 24", len(header))
+	}
+
+	if !bytes.Equal(header[0:8], []byte(ignitionHeaderKey)) {
+		return 0, 0, errors.New("Unrecognized CoreOS ISO image.")
+	}
+
+	offset := int64(binary.LittleEndian.Uint64(header[8:16]))
+	length := int64(binary.LittleEndian.Uint64(header[16:24]))
+	return offset, length, nil
+}
+
+func EmbedIgnition(inputISOPath, outputISOPath, ignitionConfig string) error {
+	coreosIgnitionHeader := make([]byte, ignitionHeaderSize)
+
+	inputISO, err := os.Open(inputISOPath)
+	if err != nil {
+		return err
+	}
+	defer inputISO.Close()
+
+	if _, err = inputISO.ReadAt(coreosIgnitionHeader, isoSystemAreaSize-ignitionHeaderSize); err != nil {
+		return err
+	}
+	embedOffset, embedLength, err := GetEmbedArea(coreosIgnitionHeader)
+	if err != nil {
+		return err
+	}
+
+	cpio, err := IgnitionImageArchive(ignitionConfig)
+	if err != nil {
+		return err
+	}
+
+	if len(cpio) > int(embedLength) {
+		return errors.Errorf("Compressed Ignition config is too large: %v > %v", len(cpio), int(embedLength))
+	}
+
+	resultISO, err := os.Create(outputISOPath)
+	if err != nil {
+		return err
+	}
+	defer resultISO.Close()
+
+	if _, err = io.Copy(resultISO, inputISO); err != nil {
+		return err
+	}
+
+	// clear out the embed area
+	embedArea := make([]byte, embedLength)
+	copy(embedArea, cpio)
+	if _, err = resultISO.WriteAt(embedArea, embedOffset); err != nil {
```

Comment:
same here, addIgnitionArchive func could be reused once merged:
https://github.com/openshift/assisted-service/pull/1151/files#diff-f6c1ca589f46921957fccf7717574392fd4c4c84936f3cf74fcc59c16e15c866R200

Reply:
My only concern with using it is that `func (e *rhcosEditor) addIgnitionArchive` requires I have an instance of an rhcosEditor which I have been trying to avoid creating.

Reply:
why not actually? i thought we could just add CreateClusterFullISO func to the isoeditor for keeping conssistency.

Reply:
It was based on a conversation with @carbonin . Even now, it seems like complete overkill for `UploadISO` to need an isoeditor + isohandler in order to embed an ignition. I'm open to be convinced otherwise but do want it to stay simple.